internal/controller/controller.go

```diff
@@ -193,6 +194,11 @@ func (s *Controller) startProcessing(actionID string) bool {
 		return false
 	}
 
+	if inProgress := len(s.startedActions); inProgress >= s.cfg.MaxActionsInProgress {
+		s.log.Warnf("too many actions in progress %d/%d", inProgress, s.cfg.MaxActionsInProgress)
+		return false
+	}
+
```

Comment:
Interesting. So the plan is to keep pulling and then check if an action can be processed.

Reply:
Is that because `PollClusterActions` returns fixed number all the time?

Reply:
I wonder if can adjust actions API and instead of silencing actions when MaxActionsInProgress exceeded try pulling less.

Reply:
Yes.`PollClusterActions` returns actions that are not completed yet. Current max is 300. CC is during the deduplication. 

There is a plan to do a redesing of the whole API but it will take more time